File: 0-99/71-simplify-path.go
package main

import "strings"

func simplifyPath(path string) string {
	var (
		segments = strings.Split(path, "/")
		stack    []string
	)
	for _, seg := range segments {
		switch seg {
		case "", ".":
			continue
		case "..":
			l := len(stack)
			if l > 0 {
				stack = stack[:l-1]
			}
		default:
			stack = append(stack, seg)
		}
	}
	return "/" + strings.Join(stack, "/")
}

File: 0-99/71-simplify-path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        segments = path.split("/")
        stack = []
        for segment in segments[1:]:
            if segment in ("", "."):
                continue
            elif segment == "..":
                if len(stack) > 0:
                    stack.pop(-1)
            else:
                stack.append(segment)
        return "/" + "/".join(stack)
